enginetest/queries/query_plans.go

```diff
@@ -926,11 +944,10 @@ inner join pq on true
 		ExpectedPlan: "Project\n" +
 			" ├─ columns: [othertable.s2, othertable.i2, mytable.i]\n" +
 			" └─ RightJoin(othertable.i2 = mytable.i)\n" +
-			"     ├─ HashLookup(child: (mytable.i), lookup: (othertable.i2))\n" +
```

Comment:
This `HashLookup` is not getting applied because of an existing bug in RightJoin planning that I uncovered in this PR. Max also found issues with RightJoins has rewritten RightJoin planning in his PR, which will fix this one and enable proper HashLookup again here.